### ISSUE TITLE
Duplicate person checking not strict enough

### ISSUE THREAD

**Issue by euph00** [euph00/pe#6]
Given that all fields are identical and name only has minor case difference, the app does not warn the user that this may be a duplicate person. This is likely to cause issues where duplicate persons are created in the system due to typo errors, or when a duplicate person is entered into the system but with the surname and given name in a different order.

This problem is further worsened as on the right display panel, the person's name is displayed in all capital letters. Alex Yeoh and alex Yeoh's detail windows look identical, and are only differentiated in the customer list as potentially duplicate entries. This problem is exacerbated with longer names where case typos may not be obvious.

![image.png](https://raw.githubusercontent.com/euph00/pe/main/files/11505096-6991-4d26-8e82-0eff10cb9e31.png)

![image.png](https://raw.githubusercontent.com/euph00/pe/main/files/541d16d1-6832-489a-9b0f-102a57243756.png)

**Comment by nus-pe-script** [comment]
# Team's Response
We believe the severity should be a low because usually a name is typed with the first letters capitalised. Only when the user attempts to create a customer with the wrong capitalisation and the customer already exists with a different capitalisation in LoyaltyLift, will this issue then occur. Hence, we believe this to be a rather rare case, fitting the definition of a low severity as stated in the PE Info.
> severity.Low : A flaw that is unlikely to affect normal operations of the product. Appears only in very rare situations and causes a minor inconvenience only.

Furthermore, the existence of these 2 customers in LoyaltyLift does not affect normal operations of the application.
# Items for the Tester to Verify
## :question: Issue severity

Team chose [`severity.Low`]
Originally [`severity.Medium`]

- [x] I disagree

**Reason for disagreement:** > We believe the severity should be a low because usually a name is typed with the first letters capitalised. Only when the user attempts to create a customer with the wrong capitalisation and the customer already exists with a different capitalisation in LoyaltyLift, will this issue then occur.

This simply restates the scenario that causes the bug without justifying why it is rare. The prevalence and cause of typo errors are discussed in this [article](https://www.wired.com/2014/08/wuwt-typos/), which presents typos to be more insidious (users are unlikely to spot the subconscious error) and common than claimed by the team. Furthermore, the team has acknowledged in their own DG that it is a "common occurrence for 2 customers to have the same full name", which is a key component of this bug:


![image.png](https://raw.githubusercontent.com/euph00/pe/main/files/087d2098-977f-4593-97dc-992fdc3b2929.png)

It can thus be argued that this bug is not as rare as they claim it to be in everyday use, especially when there are a reasonable number of records in the system. Furthermore, as discussed in the original issue, their UI design compounds the problem by making the 2 names appear the same in the display panel on the right, making it even harder to detect.


Rarity aside, on the point of severity, this bug does affect normal operations. A storeowner could easily add points and orders to the duplicate person, causing a mistake when checking how many points a customer has and when looking up a customer to see what the customer has ordered (under ORDERS on the right display). This can result in multiple errors such as mistakenly thinking a order has not been created yet when it exists under the duplicate's name, or under-counting a customer's loyalty points. Both of these cause inconvenience to the customer and user of the app.

In conclusion, this issue deserves `severity.Medium` minimally as it does cause an inconvenience that affects normal operation of the product, and the issue can appear fairly often to pose an "occasional inconvenience" where it is difficult to detect and rectify.

<catcher-end-of-segment><hr>
